### PR TITLE
[release-1.2] ✨add kubekey k3s boostrap and control plane provider

### PR DESCRIPTION
**Pull request by 24sama**
**What this PR does / why we need it**:
add kubekey k3s boostrap and control plane provider. It's a manual check-pick PR for https://github.com/kubernetes-sigs/cluster-api/pull/7528

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sbueringer** [comment]
@24sama Can you please open a cherry-pick PR for v1.3 as well? (release-1.3 branch was created on Tuesday)

We prefer to merge in order main, release-1.3, ...

**Comment by 24sama** [comment]
@sbueringer I guess you can call the bot to create a chery-pick pr on https://github.com/kubernetes-sigs/cluster-api/pull/7528

File: cmd/clusterctl/client/config_test.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ func Test_clusterctlClient_GetProvidersConfig(t *testing.T) {
 			wantProviders: []string{
 				config.ClusterAPIProviderName,
 				config.KubeadmBootstrapProviderName,
+				config.KubeKeyK3sBootstrapProviderName,
 				config.TalosBootstrapProviderName,
 				config.KubeadmControlPlaneProviderName,
```
Comment (sbueringer) [inline cmd/clusterctl/client/config_test.go:62]: Probably also needs the KubeKeyK3sBootstrapProviderName

File: docs/book/src/user/quick-start.md [hunk 2]
```diff
@@ -439,12 +439,12 @@ clusterctl init --infrastructure ibmcloud
 
 {{#/tab }}
 {{#tab KubeKey}}
+
 ```bash
 # Initialize the management cluster
 clusterctl init --infrastructure kubekey
 ```
 
-{{#/tab }}
```
Comment (sbueringer) [inline docs/book/src/user/quick-start.md:447]: We have to keep this

**Review by sbueringer — COMMENTED** [review]
two nits

/ok-to-test

**Comment by sbueringer** [comment]
Good point ;)

Looks like it works. 

> Thanks for the review. @sbueringer I guess you can call the bot to create a chery-pick pr on #7528

Good point :)

Done > https://github.com/kubernetes-sigs/cluster-api/pull/7554

File: docs/book/src/user/quick-start.md (hunk 2 above)
Comment (24sama) [reply docs/book/src/user/quick-start.md]: ok done

File: cmd/clusterctl/client/config_test.go (hunk 1 above)
Comment (24sama) [reply cmd/clusterctl/client/config_test.go:62]: done

**Comment by sbueringer** [comment]
Thx!

/approve

**Pull request merged**